packages/react-tabs/src/components/Tab/useTab.ts

```diff
@@ -15,24 +15,33 @@ import { SelectTabEvent } from '../TabList/TabList.types';
  * @param ref - reference to root HTMLElement of Tab
  */
 export const useTab_unstable = (props: TabProps, ref: React.Ref<HTMLElement>): TabState => {
-  const { content, icon, value } = props;
+  const { content, disabled: tabDisabled = false, icon, value } = props;
 
-  const { appearance, selected, onRegister, onUnregister, onSelect, size, vertical } = useContextSelector(
-    TabListContext,
-    ctx => ({
-      appearance: ctx.appearance,
-      selected: ctx.selectedValue === value,
-      onRegister: ctx.onRegister,
-      onUnregister: ctx.onUnregister,
-      onSelect: ctx.onSelect,
-      size: ctx.size,
-      vertical: !!ctx.vertical,
-    }),
-  );
+  const {
+    appearance,
+    listDisabled,
+    selectedValue,
+    onRegister,
+    onUnregister,
+    onSelect,
+    size,
+    vertical,
+  } = useContextSelector(TabListContext, ctx => ({
+    appearance: ctx.appearance,
+    listDisabled: ctx.disabled,
+    selectedValue: ctx.selectedValue,
```

Comment:
- `selectedValue` should be in a separate selector otherwise all consumers will be updated
- internally `useContextSelector()` performs **shallow** comparison with previous and a new values to perform updates (`Object.is({}, {})`), with this code it will update **always**

Reply:
@ling1726 dropped a comment (https://github.com/microsoft/fluentui/pull/22225#discussion_r838295030) with better context actually.